src/bls/bls_ies.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2018 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "bls/bls_ies.h"
+
+#include "crypto/aes.h"
+#include "hash.h"
+#include "random.h"
+#include "streams.h"
+
+template <typename Out>
+static bool EncryptBlob(const void* in, size_t inSize, Out& out, const void* symKey, const void* iv)
+{
+    out.resize(inSize);
+
+    AES256CBCEncrypt enc((const unsigned char*)symKey, (const unsigned char*)iv, false);
+    int w = enc.Encrypt((const unsigned char*)in, (int)inSize, (unsigned char*)out.data());
+    return w == (int)inSize;
```

Comment:
I wouldn't add an exception for this, as it is not "exceptional behavior". That is what the return value is for (and the caller must always check it).
If padding is disabled and the data size is not a multiple of the aes blocksize, then `AES256CBCEncrypt::Encrypt` returns 0, and this function returns false for any (non empty) object.

Also, keep in mind that this function, as well as the `CBLSIESEncryptedBlob` class, is low level. We don't access them directly, but with templated wrappers (`CBLSIESEncryptedObject<T>` / `CBLSIESMultiRecipientObjects<T>`), where the size is fixed, and given by the serialized size of the underlying object `T` (for starters a BLS secret key, which has size 32).

`CBLSIESMultiRecipientObjects` is the class we are going to use in the DKG, to exchange encrypted key shares between quorum members.
`CBLSIESEncryptedObject` is not used anywhere, actually, and I think it was added only for completeness. 
We could even consider dropping it from the codebase, and then adding it back if/when there's a specific use-case for it.

Reply:
k yeah.
it's more for an internal base class documentation to not have to re-discover this in the future in case we decide to use it for a different purpose. The not padding decision is done internally without any comment nor description.
 But, we can just add a descriptive comment at the top of the base class in the header and done for now as well.

Reply:
Added few comments.